### PR TITLE
Dala 2194 integrate eu taxonomy

### PR DESCRIPTION
**Pull request by fsoffel**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed
  - [x] The main branch is deployed to a dev server with `Reset non-user related Docker Volumes & Re-populate` turned on
  - [x] It's verified that the CD run is green
  - [x] Immediately after that the feature branch is deployed to the same server with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the dev server
- [ ] ~ELSE, the new version is deployed to the dev server "dev1" using this branch~
  - [ ] ~Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on~
  - [ ] ~It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)~
  - [ ] ~It's verified that the CD run is green~
  - [ ] ~The new feature is manually used/tested/observed on dev server~
  - [ ] ~All implemented Social Logins have been tested manually in the UI~
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        newTotalAmountObject.put("value", totalAmountValueObject)
+        cashFlowDetails.put("totalAmount", newTotalAmountObject)
+    }
+
+    private fun migrateDataPointToFinancialShare(cashFlowDetails: JSONObject, fromKey: String, toKey: String) {
+        val financialShareObject = JSONObject()
+        with(cashFlowDetails.opt(fromKey) ?: JSONObject.NULL) {
+            if (this == JSONObject.NULL) {
+                JSONObject.NULL
+            } else {
```
Comment (MarcTM01) [inline dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt:141]: Why? Der Return-Type wird doch eh nicht gebraucht.

File: dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        keysToMove.forEach {
+            generalObject.put(it, dataObject.opt(it) ?: JSONObject.NULL)
+            dataObject.remove(it)
+        }
+        dataObject.put("general", generalObject)
+    }
+
+    private fun migrateTotalAmount(cashFlowDetails: JSONObject) {
+        val totalAmountObject = cashFlowDetails.opt("totalAmount")
+        if (!invalidSet.contains(totalAmountObject) && totalAmountObject is JSONObject) {
```
Comment (MarcTM01) [inline dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt:54]: Den execution flow dieser Funktion finde ich absolut unintuitiv

File: dataland-internal-storage/src/test/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancialsTest.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.JsonUtils
+import org.json.JSONObject
```
Comment (SiegfriedSobkowiak) [inline dataland-internal-storage/src/test/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancialsTest.kt:5]: Migration seems to be broken in dev1: it appears it did not get triggered?

File: dataland-internal-storage/src/main/kotlin/db/migration/V4__MigrateEuTaxonomyNames.kt [hunk 4]
```diff
@@ -11,30 +11,34 @@ import org.json.JSONObject
...
+        "eutaxonomy-financials",
+    )
+
+    /**
+     * Migrates an old eu taxonomy  dataset to the new name
+     */
+    fun migrateEuTaxonomyNames(dataTableEntity: DataTableEntity) {
+        val companyAssociatedDatasetAsString = dataTableEntity.companyAssociatedData.toString()
+        val companyAssociatedDatasetWithEscapedSingleQuotes =
+            JSONObject(companyAssociatedDatasetAsString.replace("'", "''"))
```
[the remark this answers is not in the document]
Comment (SiegfriedSobkowiak) [reply dataland-internal-storage/src/main/kotlin/db/migration/V4__MigrateEuTaxonomyNames.kt:31]: ToDo remove as prepared statements are used

File: dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                    JSONObject.NULL
+                },
+            )
+            if (!isDataPointProvidingSourceInfo(totalAmountObject)) {
+                setAlternativeSourceInfoIfPossible(cashFlowDetails)
+            }
+        } else if (!setAlternativeSourceInfoIfPossible(cashFlowDetails)) {
+            cashFlowDetails.put("totalAmount", JSONObject.NULL)
+        } else {
+            // this else clause only exists to satisfy sonar
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-internal-storage/src/main/kotlin/db/migration/V5__MigrateToNewEuTaxonomyForNonFinancials.kt]: removed

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/eutaxonomy/nonfinancials/EuTaxonomyDetailsPerCashFlowType.kt [hunk 6]
```diff
@@ -1,17 +1,25 @@
...
  * --- API model ---
  * Fields for each cashflow type in the EuTaxonomyForNonFinancials framework
  */
 data class EuTaxonomyDetailsPerCashFlowType(
-    val totalAmount: DataPointOneValue<BigDecimal>? = null,
-
-    val alignedData: DataPointAbsoluteAndPercentage<BigDecimal>? = null,
-
-    val eligibleData: DataPointAbsoluteAndPercentage<BigDecimal>? = null,
+    val totalAmount: DataPointOneValue<AmountWithCurrency>?,
```
[the remark this answers is not in the document]
Comment (SiegfriedSobkowiak) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/eutaxonomy/nonfinancials/EuTaxonomyDetailsPerCashFlowType.kt:14]: opened backlog item for it

**Pull request merged**